src/shared/check_outdated_data.c

```diff
@@ -0,0 +1,29 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+int Run(int maxTimeDiff, char *aavild_data) {
+
+    time_t cur_time; 
+    //char * aavild_data = "20211122080000";
+    struct tm timeptr;
+
+    cur_time = time(NULL);
+    printf("Current timestamp = %ld\n", cur_time);
+
+    strptime(aavild_data, "%Y%m%d%H%M%S", &timeptr);
```

Comment:
please write a comment here describing what this is doing, as the c-code doesn't use nicely named functions it is not clear what you are doing.

Reply:
Will do sir